File: main/src/cgeo/geocaching/helper/HelperApp.java
package cgeo.geocaching.helper;

import androidx.annotation.NonNull;

final class HelperApp {
    final int titleId;
    final int descriptionId;
    final int iconId;
    @NonNull
    final String packageName;

    HelperApp(final int title, final int description, final int icon, @NonNull final String packageName) {
        this.titleId = title;
        this.descriptionId = description;
        this.iconId = icon;
        this.packageName = packageName;
    }

}

File: main-kotlin/src/main/kotlin/cgeo/geocaching/helper/HelperApp.kt
// Auto-converted from Java to Kotlin
// WARNING: This code requires manual review and likely has compilation errors
// Please review and fix:
// - Method signatures (parameter types, return types)
// - Field declarations without initialization
// - Static members (use companion object)
// - Try-catch-finally blocks
// - Generics syntax
// - Constructors
// - And more...

package cgeo.geocaching.helper

import androidx.annotation.StringRes

class HelperApp {
    final Int titleId
    final Int descriptionId
    final Int iconId
    final Int packageNameResId

    HelperApp(final Int title, final Int description, final Int icon, @StringRes final Int packageNameResId) {
        this.titleId = title
        this.descriptionId = description
        this.iconId = icon
        this.packageNameResId = packageNameResId
    }

}
